File: exercise/0-sleep.c
#include <stdio.h>
#include <unistd.h>

int main(void) {
  sleep(5);

  const char *message = "Hello from C after 5 seconds!\n";

  puts(message);

  FILE *fp = fopen("0-sleep-c.txt", "w");
  if (!fp) {
    perror("fopen");
    return 1;
  }

  fputs(message, fp);
  fclose(fp);
}

File: exercise/0-sleep.rs
use std::io::Write;

fn main() {
    std::thread::sleep(std::time::Duration::from_secs(5));

    let message = "Hello from Rust after 5 seconds!";
    println!("{}", message);

    std::fs::File::create("0-sleep-rs.txt")
        .expect("Unable to create file")
        .write_fmt(format_args!("{}\n", message))
        .expect("Unable to write to file");
}
